lib/computer.ex

```diff
@@ -3,53 +3,97 @@ defmodule Computer do
+  end
+
+  defp most_valuable_move(board, mark) do
     board
     |> Board.available_placeholders
-    |> Enum.map(fn(x) ->
-      {x, minimax(Board.set_mark(board, mark, x), mark, false, 9)} end)
-    |> Enum.max_by(fn(x) -> elem(x, 1) end)
-    |> elem(0)
+    |> try_all_moves(board, mark)
+    |> select_best_move
   end
 
-  defp minimax(board, mark) do
-    minimax(board, mark, false, 9)
+  def try_all_moves(positions_available, board, mark) do
+    Enum.map(
+      positions_available,
+      fn(position) ->
+        {position, minimax(Board.set_mark(board, mark, position), mark, false, 9)}
+      end
+    )
+  end
+
+  def select_best_move(scored_moves) do
+    scored_moves
+    |> Enum.max_by(fn(move) -> elem(move, 1) end)
+    |> elem(0)
   end
 
-  defp minimax(board, mark, computer, depth) do
+  defp minimax(board, mark, maximizing_player, depth, alpha \\ -100, beta \\ 100) do
     cond do
       Board.win?(board, mark) -> depth
       Board.win?(board, other_mark(mark)) -> -depth
       Board.tie?(board) -> 0
-      true ->
-        if computer do
-          maximizing_player(board, mark, depth)
-        else
-          minimizing_player(board, mark, depth)
-        end
+      true -> keep_playing(maximizing_player, board, mark, depth, alpha, beta)
     end
   end
 
-  defp maximizing_player(board, mark, depth) do
-    spots = board |> Board.available_placeholders
-    all_values = spots |> Enum.map(
-      fn(x) ->
-        new_board = Board.set_mark(board, mark, x)
-        minimax(new_board, mark, false, depth - 1)
+  defp keep_playing(maximizing_player, board, mark, depth, alpha, beta) do
+    if maximizing_player do
+      maximizing_player(board, mark, depth, alpha, beta)
+    else
+      minimizing_player(board, mark, depth, alpha, beta)
+    end
+  end
+
+  defp maximizing_player(board, mark, depth, alpha, beta) do
+    available_spots = Board.available_placeholders(board)
+
+    most_valuable_move = available_spots |> Enum.reduce_while(%{value_move: alpha, value_alpha: alpha},
+      fn(position, previous) ->
+        new_board = Board.set_mark(board, mark, position)
+        value_move = minimax(new_board, mark, false, depth - 1, previous.value_alpha, beta)
+
+        best_scored_move = Enum.max([value_move, previous.value_move])
+        new_alpha = Enum.max([best_scored_move, previous.value_alpha])
+
+        if (new_alpha >= beta) do
+          {:halt, %{value_move: best_scored_move, value_alpha: new_alpha}}
+        else
+          {:cont, %{value_move: best_scored_move, value_alpha: new_alpha}}
+        end
       end
     )
 
-    Enum.max(all_values)
+    most_valuable_move.value_move
   end
 
-  defp minimizing_player(board, mark, depth) do
-    spots = board |> Board.available_placeholders
-    all_values = spots |> Enum.map(
-      fn(x) ->
-        new_board = Board.set_mark(board, other_mark(mark), x)
-        minimax(new_board, mark, true, depth - 1)
+  defp minimizing_player(board, mark, depth, alpha, beta) do
```

Comment:
This duplication is pretty narly. Maybe you can extract the differences into a function that you pass in?